### DISCUSSION TITLE
DataNode TimeTick Lag Behind Now Several days

### DISCUSSION THREAD

**Discussion by xiaozhuang-a** [milvus-io/milvus#25313] [Q&A and General discussion]
![image](https://github.com/milvus-io/milvus/assets/61875985/5c2079ff-541b-412f-902a-bb08fa5b947f)
Perhaps due to the large import of collections and data, as well as certain issues, the timetick of datanode nodes has been delayed and appears to be continuously increasing.

I would like to ask, this parameter represents the delay of datanode consumption messages and how to troubleshoot and solve it

**Comment by yhmo** [comment]
There are two possible reasons for data node time tick delay:
1. intensive data insertion
If you continually insert a lot of data into Milvus, you will see the time tick delay longer and longer. Stop the insertion or reduce insert action, you will see the time tick delay become shorter.

2. too many collections
If you created thousands of collections, data nodes' time tick also can be delay.


The root cause is the Pulsar is busy to handle messive messages from Milvus.

**Reply by xiaozhuang-a** [reply]
After testing here, it was found that inserting a large number of collections does not cause problems. It only occurs when there are a large number of collections and loads. What is the reason for this? Is it because querynode consumes a large number of pulsar messages。

**Reply by xiaozhuang-a** [reply]
Thank you for your reply
Also, I would like to confirm what datanode time tick means？

**Reply by yhmo** [reply]
Milvus synchronizes data based on time sequence: https://milvus.io/docs/time_sync.md
The root coord sends a time tick message to Pulsar, and other nodes(querynode, datanode, indexnode) consume messages from Pulsar. Once a time point message is accepted by a data node, that means all the operations before this time point are already done in this data node. If a time point is accepted by datanode_A but not accepted by datanode_B, then we say some operations are not fully done. In this way, we can know what time point the data can be visible to users.

**Reply by xiaozhuang-a** [reply]
thanks。
